### PR TITLE
Add a 2nd order bc for TKE in EDMF

### PR DESCRIPTION
**Pull request by yairchn**
This PR adds a matching TKE flux at state⁺ to that in state⁻ for the TKE so that the TKE BC will have zero flux divergence and thus will prescribed a value of surface TKE as given by the EDMF model.

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [X] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Review by charleskawczynski — APPROVED** [review]
LGTM

File: test/Atmos/EDMF/edmf_kernels.jl [hunk 1]
```diff
@@ -1259,10 +1259,25 @@ function turbconv_normal_boundary_flux_second_order!(
...
+        aux = aux⁻,
+        t,
+        diffusive = diffusive⁻,
+        hyperdiffusive = hyperdiff⁻,
+    )
+    pargs = merge(_args⁻, (precomputed = precompute(atmos, _args⁻, tend_type),))
+    total_flux =
+        Σfluxes(en_ρatke(), eq_tends(en_ρatke(), atmos, tend_type), atmos, pargs)
+    nd_ρatke = dot(n⁻, total_flux)
+    en_flx.ρatke += nd_ρatke
```
Comment (ilopezgp) [inline test/Atmos/EDMF/edmf_kernels.jl:1278]: If this condition prescribes the flux at plus from the flux at minus, should this just be  `en_flx.ρatke = nd_ρatke`? Also, it  might be good to comment this intended behavior above `function turbconv_normal_boundary_flux_second_order!`

**Comment by yairchn** [comment]
bors try

**Comment by yairchn** [comment]
I have added the same BC implementation for the thermodynamic covariances - please take a look

**Comment by yairchn** [comment]
bors try

File: test/Atmos/EDMF/edmf_kernels.jl [hunk 2]
```diff
@@ -1254,15 +1254,65 @@ end;
 
 
 # The boundary conditions for second-order unknowns
+# (here we prescribe a flux at state⁺ that at state⁻ so that the flux divergence is zero)
```
Comment (ilopezgp) [inline test/Atmos/EDMF/edmf_kernels.jl:1257]: "equal to" missing somewhere

**Comment by yairchn** [comment]
Bots try

**Comment by ilopezgp** [comment]
bors try

**Comment by yairchn** [comment]
bors try

**Comment by yairchn** [comment]
bors try

**Comment by yairchn** [comment]
bors try

**Comment by yairchn** [comment]
bors try
